### PR TITLE
Making the EnvBase.action_spec property point to _input_spec["action"]

### LINKED ISSUE #405: [Feature Request] Remove `env._action_spec` and make `env.action_spec` point to `env.input_spec["action"]`
## Motivation

We keep two copies of "action_spec", one in `env._action_spec` and one in `env.input_spec["action"]`.
When applying transformations, we change both independently and they can cease to point to the same object.

We should refactor this such that `env.action_spec` always points to `env.input_spec["action"]`. When setting `env.action_spec = smth`, under the hood the env should take care of setting the action spec to `env.input_spec`.

## Solution

All refs to `env._action_spec` must be removed. 
A pointer should be implemented from `def action_spec(self)` to `self.input_spec["action"]` and from the setter to the `input_spec` setter.

In the transforms (torchrl.envs.transforms), we just need to keep the transformation of `action_spec` (not the transformation of the "action_spec" value from "input_spec"), i.e. we want the action spec to be transformed only once.

### PR DESCRIPTION
**Pull request by JulienB-78**
## Description

EnvBase._action_spec has been remove and the action_spec property now points to _input_spec["action"]

## Motivation and Context

Close #405

- [x] I have raised an issue to propose this change ([required](https://github.com/facebookresearch/rl/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/facebookresearch/rl/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: torchrl/envs/transforms/transforms.py [hunk 1]
```diff
@@ -358,21 +344,12 @@ def observation_spec(self) -> TensorSpec:
...
-        if self._action_spec is None or not self.cache_specs:
-            action_spec = self.transform.transform_action_spec(
-                deepcopy(self.base_env.action_spec)
-            )
-            if self.cache_specs:
-                self._action_spec = action_spec
-        else:
-            action_spec = self._action_spec
-        return action_spec
+        """Action spec of the transformed environment"""
```
Comment (vmoens) [inline torchrl/envs/transforms/transforms.py:347]: Do you think we still need this, as TransformedEnv inherits from EnvBase I guess the end result will be the same

File: torchrl/envs/transforms/transforms.py [hunk 2]
```diff
@@ -539,9 +516,22 @@ def __setattr__(self, key, value):
         if isinstance(value, Transform):
             value.set_parent(self)
         if isinstance(propobj, property):
-            if propobj.fset is None:
+            ancestors = list(__class__.__mro__)[::-1]
```
Comment (vmoens) [inline torchrl/envs/transforms/transforms.py:519]: I wonder what this change does? What's the reason for this?

File: torchrl/envs/transforms/transforms.py [hunk 3]
```diff
@@ -539,9 +516,22 @@ def __setattr__(self, key, value):
...
+            while isinstance(propobj, property):
+                if propobj.fset is not None:
+                    return propobj.fset(self, value)
+                propobj = getattr(ancestors.pop(), key, None)
+            else:
                 raise AttributeError(f"can't set attribute {key}")
-            return propobj.fset(self, value)
+
+        # if isinstance(propobj, property):
```
Comment (vmoens) [inline torchrl/envs/transforms/transforms.py:527]: do we want to keep this commented block?

**Review by vmoens — APPROVED** [review]
LGTM, just a couple of comments for clarification

File: torchrl/envs/transforms/transforms.py (hunk 1 above)
Comment (JulienB-78) [reply torchrl/envs/transforms/transforms.py:347]: If the logic with cache_specs is not mandatory, we could indeed removing it. As a side not this property as no setter so, I am not mistaken, action_spec cannot be be set in a TranformedEnv. (I don't think the setter is inherited if the property is redefined in the child class)

File: torchrl/envs/transforms/transforms.py (hunk 2 above)
Comment (JulienB-78) [reply torchrl/envs/transforms/transforms.py:519]: The way it was done before did not work if the property was inherited. getattr(self.__class__, key, None) return None if the property is not defined in self.__class__ but in own of its parents.

File: torchrl/envs/transforms/transforms.py (hunk 3 above)
Comment (JulienB-78) [reply torchrl/envs/transforms/transforms.py:527]: No, that should go away.

**Pull request merged**